src/MyElefant.php

```diff
@@ -43,23 +47,26 @@ class MyElefant
 
     public function __construct(){
         $this->yamlDatas = Yaml::parseFile(__DIR__.'/../MyElefant.yaml');
-        $this->dotenv = $this->initDotEnv();
-        $this->token = $this->getAuthentification(getenv('SECRET_KEY'));
-
-        if ($this->yamlDatas['APP_ENV'] == 'dev' ) {
+        if ($this->yamlDatas['APP_ENV'] == 'dev') {
             $this->error = $this->initLogger('error',__DIR__.'/../log/error.log');
             $this->info = $this->initLogger('info',__DIR__.'/../log/info.log');
         }
-
+            $this->dotenv = $this->initDotEnv();
     }
     
     /**
      * Initialize Dotenv
      * @return array
      */
     public function initDotEnv(){
```

Comment:
If it's only called in this class, it should be `private`.
A good rule of thumb is to create classes as `private`, then adjust visibility to `protected` or `public` only when it makes sense.